admin/users/views.py

```diff
@@ -263,6 +265,66 @@ def get_object(self, queryset=None):
         return serialize_user(User.load(self.kwargs.get('guid')))
 
 
+class UserWorkshopFormView(OSFAdmin, FormView):
+    form_class = WorkshopForm
+    object_type = 'user'
+    template_name = 'users/workshop.html'
+
+    def form_valid(self, form):
+        csv_file = form.cleaned_data['document']
+        final = self.parse(csv_file)
+        response = HttpResponse(content_type='text/csv')
+        response['Content-Disposition'] = 'attachment; filename="workshop.csv"'
+        writer = csv.writer(response)
+        for row in final:
+            writer.writerow(row)
+        return response
+
+    @staticmethod
+    def parse(csv_file):
```

Comment:
This method looks like it could use a little more testing. Especially in the case of Unicode, since you had to put so much work into it. If you can't come up with a good solution, at least a test that shows that unicode-filled lines get skipped properly would be fine for now.